lib/AST/ASTImporter.cpp

```diff
@@ -282,6 +282,34 @@ namespace clang {
         ToD->setImplicit();
     }
 
+    void addDeclToContexts(Decl *FromD, Decl *ToD) {
+      if (!Importer.isMinimalImport()) {
+        bool Added = false;
+        if (FromD->getDeclContext()->containsDeclAndLoad(FromD)) {
+          ToD->getDeclContext()->addDeclInternal(ToD);
+          Added = true;
+        }
+        if (ToD->getDeclContext() != ToD->getLexicalDeclContext() &&
+            FromD->getLexicalDeclContext()->containsDeclAndLoad(FromD)) {
+          ToD->getLexicalDeclContext()->addDeclInternal(ToD);
+          Added = true;
+        }
+        if (auto *FromTD = dyn_cast<TagDecl>(FromD)) {
+          auto *ToTD = cast<TagDecl>(ToD);
+          if (!Added &&
+              !FromTD->getLookupParent()->lookup(FromTD->getDeclName()).empty())
```

Comment:
`getLookupParent` finds the parent context of this context that will be used for unqualified name lookup. However, what about ADL and qualified lookup? I am afraid that we may be missing to make a decl visible in a context when the lexical and the semantic DC are different.
I think we should rather have a `lookup()` both on the result of `getParent()` and `getLexicalParent()`. If the two parents are the same then we can have a lookup only on one of them.